daceml/onnx/nodes/onnx_op.py

```diff
@@ -337,6 +338,56 @@ def validate(self, sdfg: SDFG, state: SDFGState):
                     "Expected value for required attribute '{}', got None".
                     format(attr))
 
+    def register_implementations(self):
+        """Register all implementations for this node"""
+
+        # Define own register_implementation because the one in dace attempts to instantiate the node
+        # which fails in the case of ONNX ops
+        def _register_implementation(self, name, transformation_type):
+            self.implementations[name] = transformation_type
+            match_node_name = "__" + transformation_type.__name__
+            if (hasattr(transformation_type, "_match_node")
+                    and transformation_type._match_node != match_node_name):
+                raise ValueError(
+                    "Transformation " + transformation_type.__name__ +
+                    " is already registered with a different library node.")
+            transformation_type._match_node = self
```

Comment:
Is this something we can merge into dace on its own? Why is there a need to instantiate the node in the first place?

Reply:
Here is the original method: https://github.com/spcl/dace/blob/master/dace/sdfg/nodes.py#L1187. I'm not sure how `_match_node` is used internally though

Reply:
It's used as part of the expand transformation to create the subgraph. As long as it's hashable and unique to the transformation object it should be fine.